### PR TITLE
SED-3343 support definition of autoscaling parameters in ap

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

File: step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/functions/autoscaler/AgentPoolRequirementSpec.java [hunk 1]
```diff
@@ -9,6 +11,7 @@
 public class AgentPoolRequirementSpec {
     public int numberOfAgents;
     public String agentPoolTemplateName;
+    @Schema(description = "Map of provisioning parameters. Currently only the dockerImage parameter is supported.")
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/functions/autoscaler/AgentPoolRequirementSpec.java]: This schema annotation was not used and removed in between

File: step-automation-packages/step-automation-packages-junit/pom.xml [hunk 2]
```diff
@@ -36,6 +36,13 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
         </dependency>
+        <!-- TODO: the default logback config doesn't work for some reason  -->
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-automation-packages/step-automation-packages-junit/pom.xml]: This has been removed in this branch now, but came from master

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java [hunk 3]
```diff
@@ -234,17 +235,23 @@ public ImportResult importArtefact(ExecutionContext context, Map<String, String>
...
+            // the plan accessor in context should be layered with 'inMemory' accessor on the top to temporarily store
+            // all plans from AP (in code below)
             PlanAccessor planAccessor = context.getPlanAccessor();
             if (!isLayeredAccessor(planAccessor)) {
                 result.setErrors(List.of(planAccessor.getClass() + " is not layered"));
                 return result;
             }
 
-            planAccessor.save(plan);
+            // save ALL plans from AP to the execution context to support the 'callPlan' artefact
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java]: Not sure what happened here, but the PR now looks clean

File: step-functions-plugins/step-functions-plugins-composite/step-functions-plugins-composite-def/src/test/java/step/plugins/functions/types/CompositeResolvedPlanBuilderTest.java
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-functions-plugins/step-functions-plugins-composite/step-functions-plugins-composite-def/src/test/java/step/plugins/functions/types/CompositeResolvedPlanBuilderTest.java:1]: PR looks clean now

**Review by jeromecomte — APPROVED** [review]

**Pull request merged**